akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -759,6 +760,38 @@ private[akka] class ShardRegion(
           after.map(_.address).getOrElse(""))
       coordinator = None
       startRegistration()
+    } else if (coordinator.nonEmpty) {
+      val coordAddress = coordinator.get.path.address   // safe: guarded by nonEmpty
+      val coordinatorStatus =
+        coordinator.flatMap { _ =>
+          newMembers.find(_.address == coordAddress).map(_.status)
```

Comment:
This should only rarely need to iterate deep into the member set, even on a large cluster.